src/transport/replay_transport.cpp

```diff
@@ -0,0 +1,272 @@
+
+	if (now >= record_time) {
+		/* we passed dueTime, let's program a very short relative time*/
+		msecDelta = 1;
+	} else {
+		msecDelta = std::chrono::duration_cast<std::chrono::milliseconds>(record_time - now).count();
+		if (msecDelta < 1)
+			msecDelta = 1;
+	}
+
+	timeout.it_value.tv_sec = msecDelta / 1000;
+	timeout.it_value.tv_nsec = (msecDelta % 1000) * 1000 * 1000;
+	//LOG(LOG_WARNING, "trigerring timer now sec=%ld nsec=%ld", timeout.it_value.tv_sec, timeout.it_value.tv_nsec);
+
+	if (timerfd_settime(timer, 0, &timeout, nullptr) < 0) {
+		LOG(LOG_ERR, "unable to set the timer time");
+	}
+}
+
+ReplayTransport::ReadResult ReplayTransport::read_more_chunk() {
+	uint8_t buffer[8+4+1];
+	uint32_t chunkLen, offset;
+	ReplayTransport::ReadResult status;
+
+	Transport::Read ret = inFile.atomic_read(byte_ptr(buffer), sizeof(buffer));
+	if (ret == Transport::Read::Eof) {
+		eof = true;
+		return Eof;
+	}
+
+	InStream instream(buffer, sizeof(buffer));
+
+	uint8_t recordType = instream.in_uint8();
+	record_time = start_time + std::chrono::milliseconds(instream.in_uint64_le());
+	chunkLen = instream.in_uint32_le();
+
+	switch (recordType) {
+	case RecorderTransport::RECORD_TYPE_CERT:
+		// LOG(LOG_WARNING, "cert len=%u", chunkLen);
+		public_key_size = chunkLen;
+		public_key = static_cast<uint8_t *>(malloc(chunkLen));
+		ret = inFile.atomic_read(public_key, chunkLen);
+		if (ret != Transport::Read::Ok) {
+			eof = true;
+			return Eof;
+		}
+
+		return Meta;
+		break;
+	case RecorderTransport::RECORD_TYPE_DATA_IN:
+		// LOG(LOG_WARNING, "data chunk len=%u offset=%lu", chunkLen, record_offset);
+		offset = record_ptr - record_data;
+
+		record_data = record_ptr = static_cast<char *>(realloc(record_data, record_len + chunkLen));
+		if (record_len)
+			memmove(record_ptr, record_ptr + offset, record_len);
+
+		ret = inFile.atomic_read(record_ptr + offset, chunkLen);
+		if (ret != Transport::Read::Ok) {
+			eof = true;
+			return Eof;
+		}
+		record_len += chunkLen;
+		status = Data;
+		break;
+	case RecorderTransport::RECORD_TYPE_EOF:
+		// LOG(LOG_WARNING, "EOF !!!!!!!!!!!!!!");
+		eof = true;
+		status = Eof;
+		break;
+	default:
+		break;
+	}
+
+	reschedule_timer();
+
+	return status;
+}
+
+
+const uint8_t * ReplayTransport::get_public_key() const {
+	return public_key;
+}
+
+size_t ReplayTransport::get_public_key_length() const {
+	return public_key_size;
+}
+
+Transport::TlsResult ReplayTransport::enable_client_tls(bool server_cert_store,
+                                    ServerCertCheck server_cert_check,
+                                    ServerNotifier & server_notifier,
+                                    const char * certif_path)
+{
+	(void)server_cert_store;
+	(void)server_cert_check;
+	(void)server_notifier;
+	(void)certif_path;
+
+	return Transport::TlsResult::Ok;
+}
+
+size_t ReplayTransport::do_partial_read(uint8_t * buffer, size_t len) {
+	size_t ret;
+	uint64_t timeval;
+
+	read(timer, &timeval, sizeof(timeval));
```

Comment:
/!\ Machine dependent, prefer `OutStream::out_timeval_to_uint64le_usec`/`InStream::in_timeval_from_uint64le_usec` instead.

Reply:
Nope because it's only to drain the timerfd, and what you read is always a uint64.